src/main/java/org/adoptopenjdk/jheappo/heapdump/EncodedChunk.java

```diff
@@ -0,0 +1,163 @@
+package org.adoptopenjdk.jheappo.heapdump;
+
+import java.io.PrintStream;
+import org.adoptopenjdk.jheappo.model.BasicDataTypeValue;
+import org.adoptopenjdk.jheappo.objects.BasicDataTypes;
+
+public class EncodedChunk {
```

Comment:
Javadoc please :-), explaining what U2 et are would be helpful.

Reply:
I'm not sure I can come up with a good description of what `extractU2` and friends are :(

Currently the best I can do is: "two signed bytes widened to signed ints and then shifted" which (1) might be me misunderstanding how widening conversions work in java and also (2) is pretty confusing even on a good day. Are you OK with some class-level Javadoc explaining the point of the class, and waiting on per-method ones until we have unsigned types in place? At that point it should be pretty clear -- it's a 2 byte unsigned integer, etc.

Reply:
Yep totally fine with that.